tests/multi_ptr/multi_ptr_access_members.h

```diff
@@ -85,61 +89,72 @@ class run_access_members_tests {
-          if constexpr (!std::is_void_v<multi_ptr_t::value_type>) {
+          if constexpr (!std::is_void_v<typename multi_ptr_t::value_type>) {
             // Check dereference operator return value and type correctness
             test_result.dereference_return_type_is_correct =
-                std::is_same_v<decltype(*multi_ptr), multi_ptr_t::reference>;
+                std::is_same_v<decltype(*multi_ptr),
+                               typename multi_ptr_t::reference>;
             test_result.dereference_ret_value = *multi_ptr;
             // Check operator->() return value and type correctness
             test_result.dereference_op_return_type_is_correct =
                 std::is_same_v<decltype(multi_ptr.operator->()),
-                               multi_ptr_t::pointer>;
+                               typename multi_ptr_t::pointer>;
             test_result.dereference_op_ret_value = *(multi_ptr.operator->());
           }
 
           // Check get() return value and type correctness
           test_result.get_return_type_is_correct =
-              std::is_same_v<decltype(multi_ptr.get()), multi_ptr_t::pointer>;
+              std::is_same_v<decltype(multi_ptr.get()),
+                             typename multi_ptr_t::pointer>;
           // Skip verification if pointer is decorated
           if constexpr (decorated == sycl::access::decorated::yes) {
             test_result.get_member_ret_value = *(multi_ptr.get());
           }
           // Check get_raw() return value and type correctness
-          test_result.get_raw_return_type_is_correct =
-              std::is_same_v<decltype(multi_ptr.get_raw()),
-                             std::add_pointer_t<multi_ptr_t::value_type>>;
+          test_result.get_raw_return_type_is_correct = std::is_same_v<
+              decltype(multi_ptr.get_raw()),
+              std::add_pointer_t<typename multi_ptr_t::value_type>>;
           test_result.get_raw_member_ret_value = *(multi_ptr.get_raw());
           // Check get_decorated() return type correctness
           test_result.get_decorated_return_type_is_correct =
               std::is_pointer_v<decltype(multi_ptr.get_decorated())>;
         };
 
-        if constexpr (space == sycl::access::address_space::global_space) {
-          cgh.single_task([=] { test_device_code(); });
-        } else {
-          cgh.parallel_for(sycl::nd_range(r, r), [=](sycl::nd_item item) {
-            if constexpr (space == sycl::access::address_space::local_space) {
-              test_device_code();
-            } else {
-              test_device_code();
-            }
+        if constexpr (space == sycl::access::address_space::local_space) {
+          sycl::local_accessor<T> acc_for_multi_ptr{sycl::range(1), cgh};
+          cgh.parallel_for(sycl::nd_range(r, r), [=](sycl::nd_item<1> item) {
+            value_operations::assign(acc_for_multi_ptr, value);
+            test_device_code(acc_for_multi_ptr);
```

Comment:
I think it might be worth adding a barrier between these two lines to make this code future proof. Right now it launches only one work-item, but it's easy to miss that the code doesn't work for any other work-group sizes. This applies to all places where local address space memory is initialized now.

Reply:
Good call. I have added a group-barrier on each local mem case.